dlt/destinations/dataset.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any, Generator, List, Tuple, Optional
+
+from contextlib import contextmanager
+from dlt.common.destination.reference import (
+    WithReadableRelations,
+    SupportsReadableRelation,
+    SupportsReadableDataset,
+)
+
+from dlt.common.typing import DataFrame, ArrowTable
+
+
+class ReadableRelation(SupportsReadableRelation):
+    def __init__(
+        self,
+        *,
+        client: WithReadableRelations,
+        table: str = None,
```

Comment:
we also should know the relation schema here. you'll need it to form correctly typed frames and arrow tables

Reply:
this is now implemented to querying tables